my_compassion/controllers/my2_donations.py

```diff
@@ -0,0 +1,310 @@
+        "/my2/gifts/new",
+        type="json",
+        auth="user",
+        website=True,
+        methods=["POST"],
+    )
+    def donation_new(self, **post):
+        """
+        Receives the donation submission and add it to the gift package of the user,
+        then redirects to checkout (the gift package page).
+        """
+        # Fetch the product record from the database
+        try:
+            product_template_id = int(post.get("product_id"))
+        except (ValueError, TypeError):
+            raise BadRequest()
+        product_template = (
+            request.env["product.template"].sudo().browse(product_template_id)
+        )
+
+        # Make sure the product is available
+        if not product_template.activate_for_my_compassion:
+            raise NotFound()
+
+        # Get current cart content
+        order = request.website.sale_get_order(force_create=True)
+
+        # Add product to the cart
+        order.write(
+            {
+                "order_line": [
+                    (
+                        0,
+                        0,
+                        self._extract_donation_order_line_fields(
+                            product_template, post
+                        ),
+                    )
+                ]
+            }
+        )
+
+    @http.route(
+        "/my2/gifts/edit",
+        type="json",
+        auth="user",
+        website=True,
+        methods=["POST"],
+    )
+    def donation_edit(self, **post):
+        """
+        Edits a donation from the user's gift package.
+        """
+        # Fetch the order line record from the database
+        order = request.website.sale_get_order()
+        order_line_id = post.get("order_line_id")
+        order_line = order and order.order_line.filtered(lambda l: l.id == order_line_id)
+
+        # Make sure the order line exists
+        if not order_line:
+            raise NotFound()
+
+        product_template = order_line.product_template_id
+
+        order_line.write(
+            self._extract_donation_order_line_fields(product_template, post)
+        )
+
+    @http.route(
+        "/my2/gift-package",
+        type="http",
+        auth="user",
+        website=True,
+        sitemap=False,
+    )
+    def my2_render_gift_package_page(self, **kwargs):
+        """
+        Renders the gift package (cart) page.
+        return: An HTTP response containing a rendered template with the gift package page.
+        """
+        # Get the current sales order and register it as last order
+        # (usually done in a confirmation step that we don't have)
+        order = request.website.sale_get_order()
+        request.session['sale_last_order_id'] = order.id
+
+        # Fetch gift thresholds
+        limits = request.env["gift.threshold.settings"].sudo().search([])
+
+        # Fetch PostFinance acquirer
+        postfinance_acquirer = http.request.env['payment.acquirer'].sudo().search([('provider', '=', 'postfinance')], limit=1)
```

Comment:
To me, this shouldn't be tied to PostFinance specifically. If you look at the regular /shop/payment route from Odoo, it will fetch all available providers and let the user chose which one to use. I would expect similar behavior in our checkout page. This way, the module can be used regardless of payment providers activated on a database.
```suggestion
        acquirers = http.request.env['payment.acquirer'].sudo().search([('website_published', '=', True)])
```

Reply:
Moved fetching the acquirer to a separate method to make it easier to swap it for something else in the future if needed.